tools/cache-zulip-git-version

```diff
@@ -2,4 +2,19 @@
 set -e
 
 cd "$(dirname "$0")/.."
-git describe --tags --match='[0-9]*' >zulip-git-version || true
+version=$(git describe --tags --match='[0-9]*') || "unknown"
+commit=$(git rev-parse HEAD) || "unknown"
+if git remote | grep -q "^upstream$"; then
+    merge_base_commit=$(git merge-base upstream/master HEAD) || "unknown"
+    fork_commits=$(git log --pretty=oneline "$merge_base_commit..HEAD" | wc -l) || "unknown"
+else
+    merge_base_commit="unknown"
+    fork_commits="unknown"
+fi
+
+cat >zulip-git-version <<EOF
+$version
+$commit
+$merge_base_commit
+$fork_commits
+EOF
```

Comment:
We may want to use a cleaner communication interface than a file with 4 values each on a line by itself.